src/ILCompiler.Compiler/src/Compiler/VirtualMethodCallHelper.cs

```diff
@@ -83,7 +83,11 @@ public static int GetGenericDictionarySlot(NodeFactory factory, TypeDesc type)
         /// </summary>
         public static bool HasGenericDictionarySlot(this TypeDesc type)
         {
-            return !type.IsInterface && type.HasInstantiation &&
+            // Dictionary slots on generic interfaces are necessary to support static methods on interfaces
+            if (type.IsInterface)
```

Comment:
I chatted with @davidwrighton about this. The historical reason behind making this unconditional is simplicity, and keeping method slot indices for methods on IFoo<int> and IFoo<string> identical. That won't change.